src/Loading/Plugins/babylon.babylonFileLoader.ts

```diff
@@ -269,23 +277,25 @@
                 }
                 
                 //collisions, if defined. otherwise, default is true
-                if (parsedData.collisionsEnabled != undefined) {
+                if (parsedData.collisionsEnabled !== undefined) {
                     scene.collisionsEnabled = parsedData.collisionsEnabled;
                 }
                 scene.workerCollisions = !!parsedData.workerCollisions;
 
                 var index: number;
                 var cache: number;
                 // Lights
-                for (index = 0, cache = parsedData.lights.length; index < cache; index++) {
-                    var parsedLight = parsedData.lights[index];
-                    var light = Light.Parse(parsedLight, scene);
-                    log += (index === 0 ? "\n\tLights:" : "");
-                    log += "\n\t\t" + light.toString(fullDetails);
+                if (parsedData.lights !== undefined) {
+                    for (index = 0, cache = parsedData.lights.length; index < cache; index++) {
+                        var parsedLight = parsedData.lights[index];
+                        var light = Light.Parse(parsedLight, scene);
+                        log += (index === 0 ? "\n\tLights:" : "");
+                        log += "\n\t\t" + light.toString(fullDetails);
+                    }
                 }
     
                 // Animations
-                if (parsedData.animations) {
+                if (parsedData.animations !== undefined) {

```

Comment:
All checks now are for undefined and null. I pushed the changes, but I have not done much in Github's PR system before. Do I need to do anything else to update this PR beyond pushing it?

I also wanted to mention that I did not change the check of "parsedData.autoAnimate" as I am not familiar with what that is. So I left it as it was _originally_:
`if (parsedData.autoAnimate) {`

I should also maybe mention that there are similar checks in importMesh() which I did not change as I didn't want to expand the scope of this PR too much.

I'd be happy to address both of those if  you feel it is a good idea. Thanks, I'm excited to be making my first PR to this Babylon.js!